vaadin-grid-flow-parent/vaadin-grid-testbench/src/main/java/com/vaadin/flow/component/grid/testbench/GridElement.java

```diff
@@ -395,4 +396,32 @@ private boolean isMultiselect() {
                 this);
     }
 
+    /**
+     * Select all items of the Grid.
+     */
+    public void selectAll() {
+        CheckboxElement selectAllCheckbox = $(CheckboxElement.class)
+                .id("selectAllCheckbox");
+        if (!selectAllCheckbox.isChecked()) {
+            selectAllCheckbox.click();
+        }
+    }
+
+    /**
+     * Click select all check box
+     */
+    public void clickSelectAll() {
+        CheckboxElement selectAllCheckbox = $(CheckboxElement.class)
+                .id("selectAllCheckbox");
+        selectAllCheckbox.click();
+    }
+
+    /**
+     * Deselect all items of the Grid.
+     */
+    public void deselectAll() {
+        TestBenchElement selectionCol = $("vaadin-grid-flow-selection-column")
+                .first();
+        executeScript("arguments[0].$server.deselectAll()", selectionCol);
```

Comment:
Not sure if this is what developers expect to happen when calling deselectAll through the Selenium API. This is not really an integration test then, but just calling the server API directly. I can see how this can be useful, but the implementation might be kind of misleading, and can hide problems.

Reply:
Fixed